client/homebrew/brewRenderer/toolBar/toolBar.jsx

```diff
@@ -3,23 +3,49 @@ const React = require('react');
 const { useState, useEffect } = React;
 const _ = require('lodash');
 
+import { Anchored, AnchoredBox, AnchoredTrigger } from '../../../components/Anchored.jsx';
+// import * as ZoomIcons from '../../../icons/icon-components/zoomIcons.jsx';
 
 const MAX_ZOOM = 300;
 const MIN_ZOOM = 10;
 
-const ToolBar = ({ onZoomChange, currentPage, onPageChange, totalPages })=>{
+const ToolBar = ({ onZoomChange, currentPage, onPageChange, totalPages, onStyleChange })=>{
 
 	const [zoomLevel, setZoomLevel] = useState(100);
-	const [pageNum, setPageNum]     = useState(currentPage);
+	const [pageNum, setPageNum] = useState(currentPage);
+	const [spread, setSpread] = useState('single');
+	const [startOnRight, setStartOnRight] = useState(true);
+	const [pageShadows, setPageShadows] = useState(true);
+	const [pagesStyle, setPagesStyle] = useState({});
 	const [toolsVisible, setToolsVisible] = useState(true);
+	const spreads = ['single', 'facing', 'flow'];
 
 	useEffect(()=>{
 		onZoomChange(zoomLevel);
 	}, [zoomLevel]);
 
 	useEffect(()=>{
 		setPageNum(currentPage);
-	}, [currentPage]);
+	}, [currentPage]);;
+
+	// update display arrangement when arrangement state is changed.
+	// todo: do this the 'react' way, without querying the dom.
+	useEffect(()=>{
+		const iframe = document.getElementById('BrewRenderer');
+		const pagesContainer = iframe?.contentWindow?.document.querySelector('.pages');
+
+		if(pagesContainer) {
+			spreads.forEach((spread)=>pagesContainer.classList.remove(spread));
+			pagesContainer.classList.add(spread);
+			['recto', 'verso'].forEach((leaf)=>pagesContainer.classList.remove(leaf));
+			pagesContainer.classList.add(startOnRight ? 'recto' : 'verso');
+		}
+	}, [spread, startOnRight]);
+
+	useEffect(()=>{
+		onStyleChange({ '.page': pageShadows ? {} : { boxShadow: 'none' } });
+	}, [pageShadows]);
```

Comment:
Generally, state is not intended to be used just as a trigger to put all the side effects in a `useEffect()` like this. Instead, any `onClick()` functions should call an actual handler function which applies the side effects *and* sets the new state. Then you don't need a useEffect at all.